src/main/java/com/trolley/trolley/InvoiceGateway.java

```diff
@@ -141,23 +141,23 @@ public List<Invoice> search(final SearchBy searchBy,
             case INVOICE_ID:
             case RECIPIENT_ID:
             case INVOICE_NUMBER:
+            case EXTERNAL_ID:
             case TAGS:
                 if(null == paramsList){
                     throw new InvalidFieldException("variable paramsList can not be null for the provided searchBy parameter. Refer to method's Javadoc for more details.");
                 }
-                body = "{\""+searchBy.name()+"\":"
+                body = "{\""+searchBy.getKey()+"\":"
```

Comment:
Tested locally and this is working as expected now.

Previously it was sending back a list of invoices that didn't match the external id